Hmm. tests failing on 3.6 due to clldutils looking for re.Pattern which I think was added in 3.7 (not sure why 3.7 is timing out though). Should we remove support for 3.6?

```shell
ImportError while loading conftest '/home/runner/work/python-nexus/python-nexus/tests/conftest.py'.
tests/conftest.py:5: in <module>
    from nexus import NexusReader
/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/nexus/__init__.py:4: in <module>
    from nexus.reader import NexusReader
/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/nexus/reader.py:13: in <module>
    from nexus.handlers.tree import TreeHandler
/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/nexus/handlers/tree.py:5: in <module>
    from clldutils.text import strip_brackets, split_text_with_context
/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/clldutils/text.py:130: in <module>
    def replace_pattern(pattern: typing.Union[str, re.Pattern], repl, text: str, flags=0) -> str:
E   AttributeError: module 're' has no attribute 'Pattern'
```

--- a/.github/workflows/python-package.yml
+++ b/.github/workflows/python-package.yml
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9, "3.10"]
+        python-version: [3.7, 3.8, 3.9, "3.10"]
 
     steps:
     - uses: actions/checkout@v2

--- a/setup.py
+++ b/setup.py
@@ -14,7 +14,6 @@ setup(
         "Topic :: Scientific/Engineering",
         "Topic :: Scientific/Engineering :: Bio-Informatics",
         "Topic :: Software Development :: Libraries :: Python Modules",
-        "Programming Language :: Python :: 3.6",
         "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
@@ -36,7 +35,7 @@ setup(
         ],
     },
     platforms='any',
-    python_requires='>=3.5',
+    python_requires='>=3.7',
     install_requires=[
         'newick>=1.3.1',
         'clldutils>=3.10.0',

--- a/tox.ini
+++ b/tox.ini
@@ -1,5 +1,5 @@
 [tox]
-envlist = py{36,37,38,39,310}
+envlist = py{37,38,39,310}
 skip_missing_interpreters = true
 
 [testenv]

